client/X11/xf_graphics.c

```diff
@@ -202,10 +202,16 @@ static BOOL xf_Bitmap_SetSurface(rdpContext* context, rdpBitmap* bitmap,
                                  BOOL primary)
 {
 	xfContext* xfc = (xfContext*) context;
+
+	if (!context)
+		return FALSE;
+
 	xf_lock_x11(xfc, FALSE);
 
 	if (primary)
 		xfc->drawing = xfc->primary;
+	else if (!bitmap)
```

Comment:
this check must be done some line above, or you leave the function with X11 locked. 
I'd be curious to see when that situation happens (I mean perhaps the check is to be done by the callers ?)

Reply:
Dammit, sorry, updated. Will be also checked by callers thanks to the following patch...